test/common/test_executor.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *  Author(s): Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#include <pcl/experimental/executor/executor.h>
+#include <pcl/test/gtest.h>
+
+using namespace executor;
+
+using ExecutorAlwaysAvailableTypes = ::testing::Types<const default_inline_executor,
+                                                      default_inline_executor,
+                                                      const default_sse_executor,
+                                                      default_sse_executor>;
+
+using ExecutorCondAvailableTypes = ::testing::Types<const default_inline_executor,
+                                                    default_inline_executor,
+                                                    const default_sse_executor,
+                                                    default_sse_executor,
+                                                    const default_omp_executor,
+                                                    default_omp_executor>;
+
+using ExecutorTypes = std::conditional_t<is_executor_available_v<omp_executor>,
+                                       ExecutorCondAvailableTypes,
+                                       ExecutorAlwaysAvailableTypes>;
+
+template <typename Executor>
+class ExecutorValidity : public ::testing::Test {};
+
+TYPED_TEST_SUITE(ExecutorValidity, ExecutorTypes);
+
+TYPED_TEST(ExecutorValidity, executors)
+{
+  EXPECT_TRUE((is_executor_v<TypeParam> == true));
+  EXPECT_TRUE((is_executor_v<int> == false));
+}
+
+template <typename Executor>
+class ExecutorPropertyTraits : public ::testing::Test {};
+
+TYPED_TEST_SUITE(ExecutorPropertyTraits, ExecutorTypes);
+
+TYPED_TEST(ExecutorPropertyTraits, executors)
+{
+  EXPECT_TRUE((can_require_v<TypeParam, blocking_t::always_t>));
+  EXPECT_FALSE((can_require_v<TypeParam, blocking_t::never_t>));
```

Comment:
Are the redundant parenthesis a leftover from the doctest days?

Reply:
The parenthesis is needed else it parses the `,` as a separator for arguments. As a result, it throws an error that it expects 1 argument but 2 were provided.